### ISSUE TITLE
Crash - Failed resolution of: Lcom/google/android/gms/gcm/OneoffTask$Builder;

### ISSUE THREAD

**Issue by TTransmit** [urbanairship/android-library#69]
I have made the following updates to my Android app's build.gradle (app)
```
implementation 'com.urbanairship.android:urbanairship-sdk:9.0.5'
implementation "com.google.firebase:firebase-messaging:11.8.0"
```

I'm getting the following crash on launch. 
```
03-29 17:43:13.073 12164-12182/com.bskyb.skyapptest E/AndroidRuntime: FATAL EXCEPTION: Thread-255
    Process: com.bskyb.skyapptest, PID: 12164
    java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/android/gms/gcm/OneoffTask$Builder;
        at com.urbanairship.job.GcmScheduler.scheduleJob(GcmScheduler.java:58)
        at com.urbanairship.job.GcmScheduler.schedule(GcmScheduler.java:44)
        at com.urbanairship.job.JobDispatcher.dispatch(JobDispatcher.java:105)
        at com.urbanairship.push.PushManager.init(PushManager.java:349)
        at com.urbanairship.UAirship.init(UAirship.java:681)
        at com.urbanairship.UAirship.executeTakeOff(UAirship.java:392)
        at com.urbanairship.UAirship.access$000(UAirship.java:54)
        at com.urbanairship.UAirship$2.run(UAirship.java:351)
        at java.lang.Thread.run(Thread.java:818)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.android.gms.gcm.OneoffTask$Builder" on path: DexPathList[[zip file "/data/app/com.bskyb.skyapptest-2/base.apk"],nativeLibraryDirectories=[/data/app/com.bskyb.skyapptest-2/lib/x86, /data/app/com.bskyb.skyapptest-2/base.apk!/lib/x86, /vendor/lib, /system/lib]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:511)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:469)
        at com.urbanairship.job.GcmScheduler.scheduleJob(GcmScheduler.java:58) 
        at com.urbanairship.job.GcmScheduler.schedule(GcmScheduler.java:44) 
        at com.urbanairship.job.JobDispatcher.dispatch(JobDispatcher.java:105) 
        at com.urbanairship.push.PushManager.init(PushManager.java:349) 
        at com.urbanairship.UAirship.init(UAirship.java:681) 
        at com.urbanairship.UAirship.executeTakeOff(UAirship.java:392) 
        at com.urbanairship.UAirship.access$000(UAirship.java:54) 
        at com.urbanairship.UAirship$2.run(UAirship.java:351) 
        at java.lang.Thread.run(Thread.java:818) 
    	Suppressed: java.lang.ClassNotFoundException: com.google.android.gms.gcm.OneoffTask$Builder
        at java.lang.Class.classForName(Native Method)
        at java.lang.BootClassLoader.findClass(ClassLoader.java:781)
        at java.lang.BootClassLoader.loadClass(ClassLoader.java:841)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:504)
        		... 10 more
     Caused by: java.lang.NoClassDefFoundError: Class not found using the boot class loader; no stack trace available
```

The code launching UrbanAirship is: 
```kotlin
    fun launchUrbanAirship() {
        initNotificationChannel(application)
        UAirship.takeOff(application)
        UAirship.shared().pushManager.userNotificationsEnabled = true
        val notificationFactory = CustomNotificationFactory(application)
        notificationFactory.smallIconId = R.drawable.app_icon_white
        UAirship.shared().pushManager.notificationFactory = notificationFactory
    }

    private fun initNotificationChannel(context: Context) {
        if (Build.VERSION.SDK_INT < 26) {
            return
        }
        val notificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
        val channelName = context.getString(R.string.notification_main_channel_name)
        val channel = NotificationChannel(NOTIFICATION_CHANNEL_ID,
            channelName, NotificationManager.IMPORTANCE_HIGH)
        notificationManager.createNotificationChannel(channel)
    }
```

I get the same crash with 9.0.4

**Comment by ralepinski** [comment]
Our SDK uses gcm package, not fcm.
```
    implementation "com.google.android.gms:play-services-gcm:11.8.0"
```

We are looking to switch over to the FCM package, but the APIs are slightly harder to integrate with without additional setup instructions.

**Comment by rlepinski** [comment]
You should be able to have both gcm/fcm packages installed in the same app without a problem. 8.x also uses GCM.

**Comment by TTransmit** [comment]
Adding both dependencies worked. Thanks. 
```
implementation 'com.urbanairship.android:urbanairship-sdk:9.0.5'
implementation 'com.google.firebase:firebase-messaging:11.8.0'
implementation 'com.google.android.gms:play-services-gcm:11.8.0'
```

**Comment by rlepinski** [comment]
@TTransmit We will have an update to our SDK that replaces GCM with FCM in the very near future.

**Comment by rlepinski** [comment]
@TTransmit Released in SDK 9.1.0. Here is a [migration guide](https://github.com/urbanairship/android-library/blob/master/documentation/migration/migration-guide-fcm.md).
